src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2436,12 +2436,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                                            const int fontHeight,
                                            const bool isInitialChange)
     {
+        // Don't try to inspect the core here. The Core is raising this while
+        // it's holding its write lock.
         _FontSizeChangedHandlers(fontWidth, fontHeight, isInitialChange);
     }
 
     void TermControl::_coreRaisedNotice(const IInspectable& /*sender*/,
                                         const Control::NoticeEventArgs& eventArgs)
     {
+        // Don't try to inspect the core here. The Core might be raising this

```

Comment:
nit nit: the worry is not that we will try, but that the _recipient of the event_ will try. That's why it's so... insidious. No need to refactor, but just keep in mind

Reply:
Ah okay, that makes sense. If they call back to some method on the TermControl on the same thread, and that method calls to ControlCore, we might be in danger.